awswrangler/cloudwatch.py

```diff
@@ -245,3 +245,212 @@ def read_logs(
+
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/logs.html#CloudWatchLogs.Client.filter_log_events
+
+    Note
+    ----
+    Cannot call filter_log_events with both log_stream_names and log_stream_name_prefix.
+
+    Parameters
+    ----------
+    log_group_name : str
+        The name of the log group.
+    log_stream_name_prefix : str
+        Filters the results to include only events from log streams that have names starting with this prefix.
+    log_stream_names: List[str]
+        Filters the results to only logs from the log streams in this list.
+    filter_pattern : str
+        The filter pattern to use. If not provided, all the events are matched.
+    start_time : datetime.datetime
+        Events with a timestamp before this time are not returned.
+    end_time : datetime.datetime
+        Events with a timestamp later than this time are not returned.
+    boto3_session : boto3.Session(), optional
+        Boto3 Session. The default boto3 session will be used if boto3_session receive None.
+
+    Returns
+    -------
+    pandas.DataFrame
+        Result as a Pandas DataFrame.
+
+    Examples
+    --------
+    Get all log events from log group 'aws_sdk_pandas_log_group' that have log stream prefix 'aws_sdk_pandas_log_stream'
+
+    >>> import awswrangler as wr
+    >>> df = wr.cloudwatch.filter_log_events(
+    ...     log_group_name="aws_sdk_pandas_log_group",
+    ...     log_stream_name_prefix="aws_sdk_pandas_log_stream",
+    ... )
+
+    Get all log events contains 'REPORT' from log stream
+    'aws_sdk_pandas_log_stream_one' and 'aws_sdk_pandas_log_stream_two'
+    from log group 'aws_sdk_pandas_log_group'
+
+    >>> import awswrangler as wr
+    >>> df = wr.cloudwatch.filter_log_events(
+    ...     log_group_name="aws_sdk_pandas_log_group",
+    ...     log_stream_names=["aws_sdk_pandas_log_stream_one","aws_sdk_pandas_log_stream_two"],
+    ...     filter_pattern='REPORT',
+    ... )
+
+    """
+    if log_stream_name_prefix and log_stream_names:
+        raise exceptions.InvalidArgumentCombination(
+            "Cannot call filter_log_events with both log_stream_names and log_stream_name_prefix"
+        )
+    _logger.debug("log_group_name: %s", log_group_name)
+
+    events: List[Dict[str, Any]] = []
+    if not log_stream_names:
+        describe_log_streams_args: Dict[str, Any] = {
+            "log_group_name": log_group_name,
+        }
+        if boto3_session:
+            describe_log_streams_args["boto3_session"] = boto3_session
+        if log_stream_name_prefix:
+            describe_log_streams_args["log_stream_name_prefix"] = log_stream_name_prefix
+        log_stream_names = describe_log_streams(
+            log_group_name=log_group_name, log_stream_name_prefix=log_stream_name_prefix, boto3_session=boto3_session
+        )["logStreamName"].tolist()
+    assert log_stream_names is not None
```

Comment:
I don't think we use `assert` anywhere in this codebase apart from the unit tests. It would be better to expicitely throw an error here.

However, can thus value even be `None` in this caqse? If `describe_log_streams` doesn't return any results, then wouldn't `log_stream_names` just be an empty list?

Reply:
I used `assert` to prevent `mypy` from raising `Argument 1 to "len" has incompatible type "Optional[List[str]]"; expected "Sized"`. There are some cases like mine in the codebase (lines 407 and 410 in postgresql.py) that used  `# type: ignore` to bypass the errors. but I don't like that approach, so I used `assert` instead.
I'll replace the `assert` with a `one-liner if statement` at line 434 that assigns an empty list to log_stream_names If describe_log_streams doesn't return any results.